code/modules/projectiles/projectile/beams.dm

```diff
@@ -155,6 +155,49 @@
 	impact_light_range = 2.5
 	impact_light_color_override = COLOR_LIME
 
+/obj/projectile/beam/emitter/on_hit(atom/target, blocked)
+	if(istype(target, /obj/structure/blob))
+		damage = damage * 0.25
+	. = ..()
+
+/obj/projectile/beam/emitter/drill
+	name = "driller beam"
+	icon_state = "emitter"
+	damage = 30
+	var/structural_damage = 100
+	impact_effect_type = /obj/effect/temp_visual/impact_effect/red_laser
+	light_color = COLOR_DARK_ORANGE
+
+/obj/projectile/beam/emitter/drill/on_hit(atom/target, blocked)
+	damage = initial(damage)
+	if(isobj(target) && !istype(target, /obj/structure/blob))
+		damage = structural_damage
+	if(isturf(target))
+		damage = structural_damage
+	if(istype(target, /turf/closed/mineral))
+		var/turf/closed/mineral/T = target
+		T.gets_drilled()
+	if(istype(target, /obj/structure/blob))
+		damage = damage * 0.25
```

Comment:
```suggestion
```
This is redundant with the same proc being called by the parent type, you don't need these.